closed/src/java.base/share/classes/openj9/internal/criu/SHA1PRNG.java

```diff
@@ -0,0 +1,195 @@
+ */
+
+/**
+ * This class is a modified version of the SHA1PRNG SecureRandom implementation
+ * that is found at sun.security.provider.SecureRandom.
+ * It was modified to be used by CRIUSEC NativePRNG as a mixing data source.
+ * Auto-seeding was removed, it is always seeded by NativePRNG from a
+ * blocking entropy source.
+ */
+
+public final class SHA1PRNG
+implements java.io.Serializable {
+
+    private static final long serialVersionUID = 3581829991155417889L;
+
+    // SHA-1 Digest yields 160-bit hashes which requires 20 bytes of space.
+    private static final int DIGEST_SIZE = 20;
+    private transient MessageDigest digest;
+    private byte[] state;
+    private byte[] remainder;
+    private int remCount;
+
+    SHA1PRNG(byte[] seed) {
+        init(seed);
+    }
+
+    /**
+     * This call, used by the constructor, instantiates the SHA digest
+     * and sets the seed.
+     */
+    private void init(byte[] seed) {
+
+        if (seed == null) {
+            throw new InternalError("internal error: no seed available.");
+        }
+
+        try {
+            digest = MessageDigest.getInstance("SHA-1", "CRIUSEC");
+        } catch (NoSuchProviderException | NoSuchAlgorithmException e) {
+            throw new InternalError("internal error: SHA-1 not available.", e);
+        }
+
+        engineSetSeed(seed);
+    }
+
+
+    /**
+     * Reseeds this random object. The given seed supplements, rather than
+     * replaces, the existing seed. Thus, repeated calls are guaranteed
+     * never to reduce randomness.
+     *
+     * @param seed the seed.
+     */
+    public synchronized void engineSetSeed(byte[] seed) {
+        if (state != null) {
+            digest.update(state);
+            for (int i = 0; i < state.length; i++) {
+                state[i] = 0;
+            }
+        }
+        state = digest.digest(seed);
+        remCount = 0;
+    }
+
+    private static void updateState(byte[] state, byte[] output) {
+        int last = 1;
+        boolean noCollision = false;
+
+        // state(n + 1) = (state(n) + output(n) + 1) % 2^160;
+        for (int i = 0; i < state.length; i++) {
+            // Add two bytes.
+            int stateCalc = (state[i] & 0xFF) + (output[i] & 0xFF) + last;
+            // Result is lower 8 bits.
+            byte newState = (byte)stateCalc;
+            // Store result. Check for state collision.
+            noCollision = noCollision | (state[i] != newState);
+            state[i] = newState;
+            // High 8 bits are carry. Store for next iteration.
+            last = stateCalc >>> 8;
+        }
+
+        // Make sure at least one bit changes!
+        if (!noCollision) {
+           state[0]++;
+        }
+    }
+
+
+    /**
+     * Generates a user-specified number of random bytes.
+     *
+     * @param result the array to be filled in with random bytes.
+     */
+    public synchronized void engineNextBytes(byte[] result) {
+        int index = 0;
+        int todo;
```

Comment:
This won't compile (duplicate local on line 158); please remove this and update line 176.

Reply:
done